zerver/views/streams.py

```diff
@@ -455,6 +455,11 @@ def update_subscription_properties_backend(request, user_profile, subscription_d
         if sub is None:
             return json_error(_("Not subscribed to stream id %d") % (stream_id,))
 
+        if property_converters[property] == check_bool:
+            if value == 'false':
+                value = False
+            elif value == 'true':
+                value = True
```

Comment:
this change seems wrong -- I would expect the `REQ` code above to decode this for us automatically. 

I think the bug is in your test above: you should be passing False/True to `ujson.dumps`, not "false"/"true"; similarly, in JS, one should be calling `JSON.Stringify` on `false`/`true`, not `"false"`/`"true"`